### PR TITLE
Contributing lib/ansible/modules/network/cloudengine/ce_snmp_location.py module to manage HUAWEI data center CloudEngine

### PR DESCRIPTION
**Pull request by QijunPan**
##### ISSUE TYPE

 - New Module Pull Request
##### COMPONENT NAME

ce_snmp_location.py
##### ANSIBLE VERSION

```
ansible 2.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/ansible-2.1.1.0/lib/ansible/modules/core/network/cloudengine', '/home/ansible-2.1.1.0/lib/ansible/module_utils', '/home/ansible-2.1.1.0/lib/ansible/modules', '/home/ansible-2.1.1.0', '/home/ansible-2.1.1.0/lib/ansible/utils/']
```

##### SUMMARY

```
TASK [config snmp location] ****************************************************
task path: /usr1/code/OpenNESS/code/current/KVM/intg/test/testcases/test-ce_snmp_location.yml:22
Using module file /usr/local/lib/python2.7/site-packages/ansible-2.3.0-py2.7.egg/ansible/modules/core/network/cloudengine/ce_snmp_location.py
<ce_6850> ESTABLISH LOCAL CONNECTION FOR USER: root
<ce_6850> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105 `" && echo ansible-tmp-1487053642.23-89177851915105="` echo $HOME/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105 `" ) && sleep 0'
<ce_6850> PUT /tmp/tmpZpc8Fp TO /root/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105/ce_snmp_location.py
<ce_6850> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105/ /root/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105/ce_snmp_location.py && sleep 0'
<ce_6850> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105/ce_snmp_location.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1487053642.23-89177851915105/" > /dev/null 2>&1 && sleep 0'
changed: [ce_6850] => {
    "changed": true, 
    "end_state": {
        "location": "nanjing China"
    }, 
    "existing": {}, 
    "invocation": {
        "module_args": {
            "auth_pass": null, 
            "authorize": false, 
            "host": "ce_6850", 
            "location": "nanjing China", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 12345, 
            "provider": null, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "use_ssl": false, 
            "username": "rootDC", 
            "validate_certs": true
        }, 
        "module_name": "ce_snmp_location"
    }, 
    "proposed": {
        "location": "nanjing China", 
        "state": "present"
    }, 
    "updates": [
        "snmp-agent sys-info location nanjing China"
    ]
}
```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_snmp_location.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ce_snmp_location
+version_added: "2.3"
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_location.py]: `version_added: "2.4"`

File: lib/ansible/modules/network/cloudengine/ce_snmp_location.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+short_description: Manages SNMP location configuration.
+description:
+    - Manages SNMP location configurations on CloudEngine switches.
+author:
+    - wangdezhuang (@CloudEngine-Ansible)
+options:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_location.py]: `state:` missing from DOCUMENTATION

File: lib/ansible/modules/network/cloudengine/ce_snmp_location.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+  tasks:
+
+  - name: "Config SNMP location"
+    ce_snmp_location:
+      state:  present
+      location:  nanjing China
+      provider: "{{ cli }}"
+
+  - name: "Undo SNMP location"
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_location.py]: "Undo" or "Remove"?

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by QijunPan** [comment]
@gundalow  These issues have been fixed, thank you very much.

**Comment by QijunPan** [comment]
@gundalow   These problems have been fixed， can  this PR merged?

**Review by gundalow — APPROVED** [review]

**Pull request merged**